test/lit/passes/once-reduction.wast

```diff
@@ -0,0 +1,1425 @@
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $caller-loop-1
+    ;; Add calls in a loop.
+    (loop $loop
+      (if
+        (i32.const 1)
+        (call $once)
+      )
+      (call $once)
+      (call $once)
+      (br_if $loop (i32.const 1))
+    )
+    (call $once)
+    (call $once)
+  )
+
+  ;; CHECK:      (func $caller-loop-2
+  ;; CHECK-NEXT:  (loop $loop
+  ;; CHECK-NEXT:   (if
+  ;; CHECK-NEXT:    (i32.const 1)
+  ;; CHECK-NEXT:    (call $once)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (br_if $loop
+  ;; CHECK-NEXT:    (i32.const 1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $once)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $caller-loop-2
+    ;; Add a single conditional call in a loop.
+    (loop $loop
+      (if
+        (i32.const 1)
+        (call $once)
+      )
+      (br_if $loop (i32.const 1))
+    )
+    (call $once)
+    (call $once)
+  )
+
+  ;; CHECK:      (func $caller-single
+  ;; CHECK-NEXT:  (call $once)
+  ;; CHECK-NEXT: )
+  (func $caller-single
+    ;; A short function with a single call.
+    (call $once)
+  )
+
+  ;; CHECK:      (func $caller-empty
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $caller-empty
+    ;; A tiny function with nothing at all.
+  )
+)
+
+;; Corner case: Initial value is not zero. We can still optimize this here,
+;; though in fact the function will never execute the payload call of foo(),
+;; which in theory we could further optimize.
+(module
+  ;; CHECK:      (type $none_=>_none (func))
+
+  ;; CHECK:      (import "env" "foo" (func $foo))
+  (import "env" "foo" (func $foo))
+
+  ;; CHECK:      (global $once (mut i32) (i32.const 42))
+  (global $once (mut i32) (i32.const 42))
+
+  ;; CHECK:      (func $once
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (global.get $once)
+  ;; CHECK-NEXT:   (return)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (global.set $once
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $foo)
+  ;; CHECK-NEXT: )
+  (func $once
+    (if
+      (global.get $once)
+      (return)
+    )
+    (global.set $once (i32.const 1))
+    (call $foo)
+  )
+
+  ;; CHECK:      (func $caller
+  ;; CHECK-NEXT:  (call $once)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $caller
+    (call $once)
+    (call $once)
+  )
+)
+
+;; Corner case: function is not quite once, there is code before the if, so no
+;; optimization will happen.
+(module
+  ;; CHECK:      (type $none_=>_none (func))
+
+  ;; CHECK:      (import "env" "foo" (func $foo))
+  (import "env" "foo" (func $foo))
+
+  ;; CHECK:      (global $once (mut i32) (i32.const 42))
+  (global $once (mut i32) (i32.const 42))
+
+  ;; CHECK:      (func $once
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (global.get $once)
+  ;; CHECK-NEXT:   (return)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (global.set $once
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $foo)
+  ;; CHECK-NEXT: )
+  (func $once
```

Comment:
```suggestion
  (func $not-once
```
Because it is not once, maybe rename it to something else..? (Applies for all not-once corner cases in the file)

Edit: It looks like there are too many places to fix and you use 'once' in the comments as well... So maybe nevermind.